File: learn3/learn3_ws/src/demo_cpp_nodes/src/motor_node.cpp
#include "rclcpp/rclcpp.hpp"

class MotorNode : public rclcpp::Node
{
    private:
        std::string name;
        int send_id;
        int rec_id;
    public:
        MotorNode(const std::string &name,int send_id,int rec_id) : Node(name)
        {
            this->send_id=send_id;
            this->rec_id=rec_id;
        }
        void torque_set(float torque);
};

void MotorNode::torque_set(float torque)
{
    RCLCPP_INFO(this->get_logger(),"motor_torque_set");
}

int main(int argc,char** argv)
{
    rclcpp::init(argc,argv);
    auto motor_node=std::make_shared<MotorNode>("BLDC",200,201);
    motor_node->torque_set(1.2f);
    rclcpp::spin(motor_node);
    rclcpp::shutdown();
    return 0;
}

File: learn3/learn3_ws/src/demo_py_pkg_2/demo_py_pkg_2/motor_node.py
import rclpy
from rclpy.node import Node

class MotorNode(Node):
    def __init__(self,name:str,price:float) -> None:
        self.name=name
        self.price=price
        super().__init__(name)
    def torque_output(self,torque:float):
        """
        sb 
        """
        print("fake motor")
        self.get_logger().info(f"{self.name}, torque output")

def main():
    rclpy.init()
    motor_node=MotorNode("BLDC",998.0)
    motor_node.torque_output(1.5)
    rclpy.spin(motor_node)
    rclpy.shutdown()
    